### PR TITLE
use datagram channel creator for udp sender

### PR DESCRIPTION
**Pull request by contrun**
References https://github.com/akka/akka/issues/28943

### CODE REVIEW

File: akka-actor/src/main/scala/akka/io/UdpSender.scala [hunk 1]
```diff
@@ -31,7 +28,12 @@ private[io] class UdpSender(
     with RequiresMessageQueue[UnboundedMessageQueueSemantics] {
 
   val channel = {
-    val datagramChannel = DatagramChannel.open
+    val datagramChannel = options
+      .collectFirst {
+        case creator: DatagramChannelCreator => creator
+      }
+      .getOrElse(DatagramChannelCreator())
+      .create()
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/io/UdpSender.scala:36]: And `DatagramChannelCreator.create` in turn calls `open`. Makes sense to me.

**Review by johanandren — APPROVED** [review]
LGTM

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
Thank you for this fix!

**Pull request merged**